agent_api/src/main/java/dev/aikido/agent_api/collectors/WebRequestCollector.java

```diff
@@ -19,20 +19,25 @@ public record Res(String msg, Integer status) {};
      * This function gets called in the initial phases of a request.
      * @param newContext is the new ContextObject that holds headers, query, ...
      * Also returns a response depending on blocking status :
-     * 1) First check: IP allowed to access route (restricted /admin routes e.g.)
-     * 2) Bypassed IP? If bypassed, no further checks
+     * 1) If the IP is bypassed we just won't set the context object
+     * 2) IP allowed to access route (restricted /admin routes e.g.)
      * 3) Blocked IP Lists: e.g. geo-blocking, crowdsec, ...
      * 4) UA Blocking: e.g. bot blocking, AI scraper blocking, ...
      */
     public static Res report(ContextObject newContext) {
```

Comment:
I find method name `report` a bit confusing, would expect something like `onRequest`?

Reply:
it's a unified thing across all the collector classes